internal/rootcoord/show_partition_task.go

```diff
@@ -57,6 +58,9 @@ func (t *showPartitionTask) Execute(ctx context.Context) error {
 	}
 
 	for _, part := range coll.Partitions {
+		if part.State != etcdpb.PartitionState_PartitionCreated {
```

Comment:
It seems that partition will be removed from meta after DropPartition:
![image](https://github.com/milvus-io/milvus/assets/42060877/192e3dda-8ace-4801-9289-25b2bdb8eafa)